core/ecschema-metadata/src/Metadata/Format.ts

```diff
@@ -107,43 +130,80 @@ export class Format extends SchemaItem {
       if (formatProps.composite.units.length <= 0 || formatProps.composite.units.length > 4)
         throw new ECSchemaError(ECSchemaStatus.InvalidECJson, `The Format ${this.fullName} has an invalid 'Composite' attribute. It should have 1-4 units.`);
     }
+
+    // For Ratio formats: validate that either composite or ratioUnits is provided
+    if (this.type === FormatType.Ratio) {
+      const hasComposite = undefined !== formatProps.composite && formatProps.composite.units.length > 0;
+      const hasRatioUnits = undefined !== formatProps.ratioUnits && formatProps.ratioUnits.length === 2;
+      if (!hasComposite && !hasRatioUnits) {
+        throw new ECSchemaError(ECSchemaStatus.InvalidECJson, `The Format ${this.fullName} is 'Ratio' type and must have either 'composite' units or 'ratioUnits'.`);
+      }
+    }
   }
 
   public override fromJSONSync(formatProps: SchemaItemFormatProps) {
     super.fromJSONSync(formatProps);
     this.typecheck(formatProps);
-    if (undefined === formatProps.composite)
-      return;
-
-    // Units are separated from the rest of the deserialization because of the need to have separate sync and async implementation
-    for (const unit of formatProps.composite.units) {
-      const newUnit = this.schema.lookupItemSync(unit.name);
-      if (undefined === newUnit || (!Unit.isUnit(newUnit) && !InvertedUnit.isInvertedUnit(newUnit)))
-        throw new ECSchemaError(ECSchemaStatus.InvalidECJson, ``);
-
-      if(Unit.isUnit(newUnit))
-        this.addUnit(new DelayedPromiseWithProps(newUnit.key, async () => newUnit), unit.label);
-      else if(InvertedUnit.isInvertedUnit(newUnit))
-        this.addUnit(new DelayedPromiseWithProps(newUnit.key, async () => newUnit), unit.label);
+
+    // Process composite units
+    if (undefined !== formatProps.composite) {
+      for (const unit of formatProps.composite.units) {
+        const newUnit = this.schema.lookupItemSync(unit.name);
+        if (undefined === newUnit || (!Unit.isUnit(newUnit) && !InvertedUnit.isInvertedUnit(newUnit)))
+          throw new ECSchemaError(ECSchemaStatus.InvalidECJson, ``);
+
+        if (Unit.isUnit(newUnit))
+          this.addUnit(new DelayedPromiseWithProps(newUnit.key, async () => newUnit), unit.label);
+        else if (InvertedUnit.isInvertedUnit(newUnit))
+          this.addUnit(new DelayedPromiseWithProps(newUnit.key, async () => newUnit), unit.label);
+      }
+    }
+
+    // Process ratioUnits
+    if (undefined !== formatProps.ratioUnits && formatProps.ratioUnits.length === 2) {
```

Comment:
This looks like a very similar, if not exactly the same block as where you process ratioUnits below lines 196-206... please extract?